File: frameworks/test.persistence/src/main/java/com/example/scaffold/frameworks/test/persistence/PersistenceExtension.java
package com.example.scaffold.frameworks.test.persistence;

import org.junit.jupiter.api.extension.BeforeEachCallback;
import org.junit.jupiter.api.extension.ExtensionContext;

import java.util.List;

public class PersistenceExtension implements BeforeEachCallback {

    @Override
    public void beforeEach(ExtensionContext context) {
        List<Object> instances = context.getRequiredTestInstances().getAllInstances();
        new Instances(instances).autoAssignPersistence();
    }

}

File: frameworks/test.persistence/src/main/kotlin/com/example/scaffold/frameworks/test/persistence/PersistenceExtension.kt
package com.example.scaffold.frameworks.test.persistence

import org.junit.jupiter.api.extension.BeforeEachCallback
import org.junit.jupiter.api.extension.ExtensionContext

class PersistenceExtension : BeforeEachCallback {
    override fun beforeEach(context: ExtensionContext) {
        val instances = context.requiredTestInstances.allInstances
        Instances(instances).autoAssignPersistence()
    }
}
